tests/javascript_tests/test_shape_controller.js

```diff
@@ -0,0 +1,145 @@
+requirejs([
+    "jquery",
+    "underscore",
+    "model",
+    "view",
+    "viewcontroller",
+    "three",
+    "shape-controller",
+    "shape-editor",
+    "shapes"
+], function($, _, model, DecompositionView, viewcontroller, THREE,
+            ShapeController, ShapeEditor, shapes) {
+  $(document).ready(function() {
+    var EmperorAttributeABC = viewcontroller.EmperorAttributeABC;
+    var DecompositionModel = model.DecompositionModel;
+
+    module("Shape Controller", {
+      setup: function() {
+        // setup function
+        shapesAvailable = ['sphere', 'cube', 'cone', 'icosahedron', 'cylinder'];
+        this.sharedDecompositionViewDict = {};
+        var $slickid = $('<div id="fooligans"></div>');
+        $slickid.appendTo(document.body);
+
+        // setup function
+        name = "pcoa";
```

Comment:
Seems like we are leaking `name`, `ids`, etc to the global scope, can you "localify" them or add them to `this`?